src/commands.ts

```diff
@@ -17,14 +22,28 @@ import {
   setAttr,
 } from './util';
 import { tableNodeTypes } from './schema';
-
-// Helper to get the selected rectangle in a table, if any. Adds table
-// map, table node, and table start offset to the object for
-// convenience.
-export function selectedRect(state) {
-  let sel = state.selection,
+import { Direction } from './input';
+
+/**
+ * @public
+ */
+export type TableRect = Rect & {
+  tableStart: number;
+  map: TableMap;
+  table: Node;
+};
```

Comment:
I think `TableRect` shouldn't extends `Rect` in the first place (i.e. in the `index.d.ts` before this PR). None of the place that use `TableRect` is using the attributes in `Rect`. Maybe we can correct this mistake by removing `Rect &` here?

Reply:
`Rect &` is required, or `yarn typecheck` would with 40 errors. One example of usage amongst many is:


```ts
export function addColumnAfter(
  state: EditorState,
  dispatch?: (tr: Transaction) => void,
): boolean {
  if (!isInTable(state)) return false;
  if (dispatch) {
    const rect = selectedRect(state);
    dispatch(addColumn(state.tr, rect, rect.right));   <=== HERE WE ALSO NEED Rect
  }
  return true;
}
```

Reply:
I see. Let's keep `Rect &` then.